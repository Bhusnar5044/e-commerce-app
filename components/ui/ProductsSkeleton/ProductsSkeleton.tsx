import { FC, memo } from 'react';
import Skeleton from '../Skeleton';

export const ProductsSkeleton: FC = memo(() => {
    return (
        <div className="mt-8 grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-4">
            <Skeleton className="aspect-square rounded-xl" />
            <Skeleton className="aspect-square rounded-xl" />
            <Skeleton className="aspect-square rounded-xl" />
            <Skeleton className="aspect-square rounded-xl" />
            <Skeleton className="aspect-square rounded-xl" />
            <Skeleton className="aspect-square rounded-xl" />
        </div>
    );
});

ProductsSkeleton.displayName = 'ProductsSkeleton';
