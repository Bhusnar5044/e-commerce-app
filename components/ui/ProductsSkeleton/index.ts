export { ProductsSkeleton as default } from './ProductsSkeleton';
