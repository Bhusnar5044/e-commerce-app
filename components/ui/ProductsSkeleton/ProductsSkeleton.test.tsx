import { render } from '@test-utils';
import { ProductsSkeleton } from './ProductsSkeleton';

describe('ProductsSkeleton', () => {
    it('should render properly', () => {
        const { container } = render(<ProductsSkeleton />);
        expect(container).toMatchSnapshot();
    });
});
