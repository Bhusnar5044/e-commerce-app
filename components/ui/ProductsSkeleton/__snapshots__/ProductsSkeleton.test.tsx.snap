// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsSkeleton should render properly 1`] = `
<div>
  <div
    class="mt-8 grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-4"
  >
    <div
      class="animate-pulse bg-neutral-200 aspect-square rounded-xl"
    />
    <div
      class="animate-pulse bg-neutral-200 aspect-square rounded-xl"
    />
    <div
      class="animate-pulse bg-neutral-200 aspect-square rounded-xl"
    />
    <div
      class="animate-pulse bg-neutral-200 aspect-square rounded-xl"
    />
    <div
      class="animate-pulse bg-neutral-200 aspect-square rounded-xl"
    />
    <div
      class="animate-pulse bg-neutral-200 aspect-square rounded-xl"
    />
  </div>
</div>
`;
